### PR TITLE
added option "caching_enabled" to vmware

### PR DESCRIPTION
**Pull request by IncredibleRichie**
Added option "caching_enabled" to vmware

### CODE REVIEW

File: plugins/modules/compute_resource.py [hunk 1]
```diff
@@ -173,6 +173,7 @@
       - ACME
     provider: vmware
     provider_params:
+      caching_enabled: 0
```
Comment (evgeni) [inline plugins/modules/compute_resource.py]: If it's `bool`, it's probably better to have the example say `caching_enabled: False`

File: plugins/modules/compute_resource.py [hunk 2]
```diff
@@ -360,6 +362,7 @@ def main():
                 tenant=dict(),
                 app_ident=dict(),
                 datacenter=dict(),
+                caching_enabled=dict(type='bool'),
```
Comment (evgeni) [inline plugins/modules/compute_resource.py:369]: You'll have to add that to the DOCUMENTATION string in the suboptions of `provider_params`:
https://github.com/theforeman/foreman-ansible-modules/blob/15db065de93584b561aec4bee3990aaf3341ac23/plugins/modules/compute_resource.py#L54-L58

See `use_v4` or `ssl_verify_peer` for examples how to do this :)

**Review by evgeni — CHANGES_REQUESTED** [review]

File: plugins/modules/compute_resource.py (hunk 1 above)
Comment (IncredibleRichie) [reply plugins/modules/compute_resource.py]: You´re right. Applied

File: plugins/modules/compute_resource.py (hunk 2 above)
Comment (IncredibleRichie) [reply plugins/modules/compute_resource.py:369]: Added

**Comment by mdellweg** [comment]
I believe the new process requires this to be escorted by a changelog entry.
You should find examples here: https://github.com/theforeman/foreman-ansible-modules/tree/develop/changelogs/fragments

@evgeni Is this even correct? Can we provide a contributing section in the docs?

**Comment by evgeni** [comment]
Correct. And yes, I need to do docs and also a check in CI (I have a discussion about that running upstream currently)

**Comment by evgeni** [comment]
Thanks @IncredibleRichie! The diff looks good to me!

As Matthias pointed out, this should get an changelog entry. Could you please add a file in `changelogs/fragments` (I'd suggest smth like `changelogs/fragments/939-compute_resource-caching_enabled.yaml`) that describes the change? The following is probably a good start:

```yaml
minor_changes:
  - compute_resource - added ``caching_enabled`` option for VMware compute resources
```

**Comment by IncredibleRichie** [comment]
@evgeni The comment came just in time, i have added the suggestions to the fragment. Thanks for the link @mdellweg

**Review by evgeni — APPROVED** [review]

**Pull request merged**